internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olm
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"text/template"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/cluster-api/util/container"
+)
+
+const (
+	defaultIndexImage        = "quay.io/operator-framework/upstream-opm-builder:latest"
+	defaultContainerName     = "registry-grpc"
+	defaultContainerPortName = "grpc"
+	defaultGRPCPort          = 50051
+	semverBundleAddMode      = "semver"
+	replacesBundleAddMode    = "replaces"
+)
+
+// RegistryPod holds resources necessary for creation of a registry server
+type RegistryPod struct {
+	// BundleAddMode specifies the graph update mode that defines how channel graphs are updated
+	BundleAddMode string
+
+	// BundleImage specifies the container image that opm uses to generate and incrementally update the database
+	BundleImage string
+
+	// Index image contains a database of pointers to operator manifest content that is queriable via an API.
+	// new version of an operator bundle when published can be added to an index image
+	IndexImage string
+
+	// DBPath refers to the registry DB;
+	// if an index image is provided, the existing registry DB is located at /database/index.db
+	DBPath string
+
+	// Namespace refers to the specific namespace in which the registry pod will be created and scoped to
+	Namespace string
+
+	// Kubeclient refers to a Kubernetes clientset that implements kubernetes.Interface.
+	Kubeclient kubernetes.Interface
+
+	// GRPCPort is the container grpc port which is defaulted to 50051
+	GRPCPort int32
+}
+
+// newRegistryPod initializes the RegistryPod struct and sets defaults for empty fields
+func newRegistryPod(kubeclient kubernetes.Interface, dbPath, bundleImage, namespace string) *RegistryPod {
+	rp := &RegistryPod{}
+
+	if rp.GRPCPort == 0 {
+		rp.GRPCPort = defaultGRPCPort
+	}
+
+	if len(rp.IndexImage) == 0 {
```

Comment:
Nit:
```suggestion
	if rp.IndexImage == "" {
```
ditto elsewhere